lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java

```diff
@@ -145,26 +159,43 @@ TimestampBounds getTimestampBounds() {
     }
 
     private void checkTimestamp(
-            TimestampBounds bounds, OperationType type, long lowerFreshTimestamp, long upperFreshTimestamp) {
+            TimestampBounds bounds,
+            OperationType type,
+            long lowerFreshTimestamp,
+            long upperFreshTimestamp,
+            Instant timeOfRequest) {
+        timestampBoundRecordHistory.put(
+                timeOfRequest, ImmutableTimestampBoundsRecord.of(bounds, type, lowerFreshTimestamp, timeOfRequest));
```

Comment:
Do we want to add to the history on failed attempts? My instinct says that we shouldn't (since we should log enough about the failed attempt to not need it in the bounds too).